include/boost/dll/smart_library.hpp

```diff
@@ -0,0 +1,189 @@
+/*
+ * smart_library.hpp
+ *
+ *  Created on: 17.10.2015
+ *      Author: Klemens
+ */
+
+#ifndef INCLUDE_BOOST_DLL_SMART_LIBRARY_HPP_
+#define INCLUDE_BOOST_DLL_SMART_LIBRARY_HPP_
+
+#include <boost/dll/shared_library.hpp>
+#include <boost/dll/mangled_storage.hpp>
+#include <boost/dll/detail/get_mem_fn_type.hpp>
+#include <boost/dll/detail/mem_fn_cast.hpp>
+#include <boost/dll/detail/ctor_dtor.hpp>
+
+#include <iostream>
```

Comment:
Do not include `iostream` in header files without big need. Try not including it at all, or including `iosfwd`